homeassistant/components/n26/sensor.py

```diff
@@ -0,0 +1,294 @@
+
+The sensor allows to show
+* N26 account balance
+* Spaces (sub-accounts within an account) balance
+* bank card status
+"""
+from datetime import timedelta
+import logging
+
+from homeassistant.components.n26 import (
+    ACCOUNT_INFO, API, CARDS, DOMAIN as COMPONENT_DOMAIN, SPACES,
+    timestamp_ms_to_date)
+from homeassistant.helpers.entity import Entity
+
+DEPENDENCIES = ['n26']
+_LOGGER = logging.getLogger(__name__)
+
+SCAN_INTERVAL = timedelta(minutes=10)
+
+ATTR_IBAN = "account"
+ATTR_USABLE_BALANCE = "usable_balance"
+ATTR_AVAILABLE_BALANCE = "available_balance"
+
+ATTR_ACC_OWNER_TITLE = "owner_title"
+ATTR_ACC_OWNER_FIRST_NAME = "owner_first_name"
+ATTR_ACC_OWNER_LAST_NAME = "owner_last_name"
+ATTR_ACC_OWNER_GENDER = "owner_gender"
+ATTR_ACC_OWNER_BIRTH_DATE = "owner_birth_date"
+ATTR_ACC_OWNER_EMAIL = "owner_email"
+ATTR_ACC_OWNER_PHONE_NUMBER = "owner_phone_number"
+
+ICON_ACCOUNT = 'mdi:currency-eur'
+ICON_CARD = 'mdi:credit-card'
+ICON_SPACE = 'mdi:crop-square'
+
+
+async def async_setup_platform(
+        hass, config, async_add_entities, discovery_info=None):
+    """Set up the N26 sensor platform."""
+    api = hass.data[COMPONENT_DOMAIN][API]
+    account_info = hass.data[COMPONENT_DOMAIN][ACCOUNT_INFO]
+    balance = api.get_balance()
+    limits = api.get_account_limits()
+    account_statuses = api.get_account_statuses()
+
+    sensor_entities = [N26Account(
+        api, account_info, account_statuses, limits, balance)]
+
+    cards = hass.data[COMPONENT_DOMAIN][CARDS]
+    for card in cards:
+        sensor_entities.append(N26Card(api, balance["iban"], card))
+
+    spaces = hass.data[COMPONENT_DOMAIN][SPACES]
+    for space in spaces["spaces"]:
+        sensor_entities.append(N26Space(api, space))
+
+    async_add_entities(sensor_entities)
+
+
+class N26Account(Entity):
+    """Sensor for a N26 balance account.
+
+    A balance account contains an amount of money (=balance). The amount may
+    also be negative.
+    """
+
+    def __init__(self, client, account_info,
+                 account_statuses, limits, balance) -> None:
+        """Initialize a N26 balance account."""
+        self._client = client
+        self._account_info = account_info
+        self._account_statuses = account_statuses
+        self._limits = limits
+        self._balance = balance
+
+    @property
+    def should_poll(self) -> bool:
+        """Return True.
+
+        Data needs to be polled from the bank servers.
+        """
+        return True
+
+    def update(self) -> None:
+        """Get the current balance and currency for the account."""
+        self._account_info = self._client.get_account_info()
+        self._account_statuses = self._client.get_account_statuses()
+        self._limits = self._client.get_account_limits()
+        self._balance = self._client.get_balance()
+
+    @property
+    def unique_id(self):
+        """Return the unique ID of the entity."""
+        return self._balance["iban"]
```

Comment:
None check? I would prefer save iban in the `__init__` method.

I assume iban is the full account number, isn't it? Shall we only use last four digital as part of the unique_id?
In US, we treat account number is sensitive information, the practice here is only display part of account number.

Reply:
Yes that is done here often as well.
It might be sufficient to only use the last 4 digits, but whats the point of hiding it if the user that configured this component can see it anyway?